src/providers/gdal/qgsgdalprovider.h

```diff
@@ -211,6 +211,12 @@ class QgsGdalProvider : public QgsRasterDataProvider, QgsGdalProviderBase
 
     //! \brief sublayers list saved for subsequent access
     QStringList mSubLayers;
+
+    //! Whether a per-dataset mask band is exposed as an alpha band for the point of view of the rest of the application.
+    bool mMaskBandExposedAsAlpha;
```

Comment:
> I'm guessing this will fix the issue we discussed briefly at the Girona hackfests?

Probably. Can't already remember ;-)

> I'm getting used to c++11 style initializers and seeing uninitisalised bools like this makes me uncomfortable... Could you move the initialisation here?

Fixed